packages/terra-paginator/src/Paginator.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import IconNext from 'terra-icon/lib/icon/IconNext';
+import IconPrevious from 'terra-icon/lib/icon/IconPrevious';
+import ResponsiveElement from 'terra-responsive-element';
+
+import 'terra-base/lib/baseStyles';
+import styles from './Paginator.scss';
+
+import { calculatePages, pageSet } from './_paginationUtils';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Function to be executed when a navigation element is selected.
+   */
+  onPageChange: PropTypes.func.isRequired,
+  /**
+   * The active/selected page.
+   */
+  selectedPage: PropTypes.number,
+  /**
+   * Total number of all items being paginated.
+   */
+  totalCount: PropTypes.number,
+  /**
+   * Total number of items per page.
+   */
+  itemCountPerPage: PropTypes.number,
+};
+
+class Paginator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { selectedPage, totalCount, itemCountPerPage } = this.props;
+
+    this.state = {
+      selectedPage: selectedPage && totalCount ? selectedPage : undefined,
+      pageSequence: selectedPage && totalCount ? pageSet(selectedPage, calculatePages(totalCount, itemCountPerPage)) : undefined,
+    };
+
+    this.handlePageChange = this.handlePageChange.bind(this);
+    this.hasNavContext = this.hasNavContext.bind(this);
+    this.buildPageButtons = this.buildPageButtons.bind(this);
+    this.reducedPaginator = this.reducedPaginator.bind(this);
+  }
+
+  handlePageChange(index) {
+    return (event) => {
```

Comment:
In the War room on Friday, we discussed not attaching the events to handle functions like this, and only using specific parameters, such as page number in this instance. Are we wanting to implement that here?

Reply:
https://github.com/cerner/terra-core/pull/1445/commits/66aa195a4cc92751d1eaa2877d8800551ae03270